File: Backend/mapping/src/main/java/com/mhp/coding/challenges/mapping/mappers/ArticleMapper.kt
package com.mhp.coding.challenges.mapping.mappers

import com.mhp.coding.challenges.mapping.models.db.Article
import com.mhp.coding.challenges.mapping.models.dto.ArticleDto
import org.springframework.stereotype.Component

@Component
class ArticleMapper {
    fun map(article: Article?): ArticleDto {
        //TODO
        return ArticleDto()
    }

    // Nicht Teil dieser Challenge.
    fun map(articleDto: ArticleDto?): Article = Article()
}

File: Backend/mapping-dotnet/MHP.CodingChallenge.Backend.Mapping.Data/ArticleMapper.cs
using System;
using System.Collections.Generic;
using System.Linq;
using MHP.CodingChallenge.Backend.Mapping.Data.DB;
using MHP.CodingChallenge.Backend.Mapping.Data.DTO;
using AutoMapper;

namespace MHP.CodingChallenge.Backend.Mapping.Data
{
    public class ArticleMapper
    {
        public ArticleDto Map(Article article)
        {
            //TODO
            return new ArticleDto
            {
                Author = article.Author,
                Description = article.Description,
                Id = article.Id,
                Title = article.Title,
                Blocks = new List<ArticleBlockDto>()
            };
        }

        public Article Map(ArticleDto articleDto)
        {
            // Nicht Teil dieser Challenge.
            return new Article();
        }

    }
}
